src/main/java/com/starinfo/StarInfoPlugin.java

```diff
@@ -618,6 +645,36 @@ public void refresh()
 		refreshHintArrow();
 	}
 
+	private void updateBonusCounter()
+	{
+		String tooltip = ColorUtil.wrapWithColorTag("Star Discovery", new Color(255, 119, 0)) + "</br>"
+				+ ColorUtil.wrapWithColorTag("Bonus stardust: ", Color.YELLOW) + bonusCount;
+
+		if (bonusCounter != null)
+		{
+			bonusCounter.setCount(bonusCount);
+			bonusCounter.setTooltip(tooltip);
+			return;
+		}
+
+		removeBonusCounter();
+
+		if (bonusCount == 0)
+		{
+			return;
+		}
+
+		bonusCounter = new BonusCounter(itemManager.getImage(STARDUST, 175, false), this, bonusCount);
```

Comment:
The image shouldn't be created on demand with every update to the counter since it's a semi expensive operation. Instead it should be created just once and cached, maybe on the plugin start up or on login if it is null if that doesn't work.